src/twisted/plugins/cred_unix.py

```diff
@@ -65,7 +68,9 @@ def checkPwd(self, pwd, username, password):
 
     def checkSpwd(self, spwd, username, password):
         try:
-            cryptedPass = spwd.getspnam(username)[1]
+            if not isinstance(username, StringType):
+                username = username.decode('utf-8')
+            cryptedPass = spwd.getspnam(username).sp_pwdp
```

Comment:
@rodrigc Sorry for jumping in randomly here, but I'd looked at this for a moment.  Since you have a PR open I thought I'd help however I could.

Looks like the sp_pwd attribute is different between python 2/3.  `sp_pwd` vs `sp_pwdp`
https://docs.python.org/3.5/library/spwd.html
https://docs.python.org/2/library/spwd.html

Which gives me the following when I run on your branch with p2 (exception only)
```
	  File "/home/jlitzing/dev/twisted/twisted-fork/src/twisted/internet/defer.py", line 150, in maybeDeferred
	    result = f(*args, **kw)
	  File "/home/jlitzing/dev/twisted/twisted-fork/src/twisted/plugins/cred_unix.py", line 100, in requestAvatarId
	    checked = self.checkSpwd(spwd, username, password)
	  File "/home/jlitzing/dev/twisted/twisted-fork/src/twisted/plugins/cred_unix.py", line 73, in checkSpwd
	    cryptedPass = spwd.getspnam(username).sp_pwdp
	exceptions.AttributeError: 'spwd.struct_spwd' object has no attribute 'sp_pwdp'
```

Reply:
Fixed.